### DISCUSSION TITLE
Extending VLANs in Dual-DC-L3LS scenario.

### DISCUSSION THREAD

**Discussion by rohandec1980** [aristanetworks/avd#3999] [Data Model Schemas] [answered]
We have a EVPN dual DC L3LS scenario, with dedicated evpn gateways deployed in each DC. 

1. Separate VLAN ranges are deployed in each DC.  Only a few layer 2 VLANs are stretched between DCs.
2. A dedicated pair of leaf switches are used for deploying evpn gateway function as well as DCI link termination.

In this scenario using the AVD example how can i achieve the following:

1. Disable the extension of all VLANs to both DCs.
2. Disable the SVI creation on leaf switches where a VLAN is not deployed. eg:evpn gateways
3. Selectively extend only the required VLANs between both DCs

Regards
Datta

**Comment by ClausHolbechArista** [comment]
Hi Datta

Re 1. & 3.
Currently the gateway feature implementation in AVD only supports extending all VLANs defined on the gateway switch, so you can leverage the `tags` defined under `svis`, `l2vlans` in combination with `filter` - `tags` under the nodes. The default tags both places are `["all"]`.

So if you set the tags on the svis/l2vlans to `["all", "evpn-gw"]` and set the tags on the gateways to `["evpn-gw"]` it might do what you need.

Re 2.
I am not sure the EVPN GW feature it works as it should without the SVIs. Please work with your Arista account team to ensure the design is following the best practices. I may be missing your point here, so feel free to elaborate.

Let me know how it works out.

Thanks
Claus

**Reply by rohandec1980** [reply]
Hi Claus

Thanks for the reply, I created 2 tags DC1-VLANs and DC2-VLANs. In the leaf nodes for DC1 I applied filter tags: DC1-VLANs. This pruned the DC2-VLANs from DC1 leaf switches.

About point 2, in our deployment we have the evpn-gateways as dedicated switches. They don't have servers connected. in this case would they still need the VLAN SVI. We wanted the vlans which need to be extended between the DCs to be configured on the gateways (without the SVIs) to allow the advertisement of EVPN type-2 prefixes between the evpn gateways between the 2 DCs. Is it possible to implement this setup with the existing AVD keys?

Is there an Arista design best practices guide/document around gateways?

Thanks again,
Datta

**Reply by ClausHolbechArista** [reply, marked as answer]
@rohandec1980 you can try to consult the new Arista Tech Library. I only found examples including the SVIs. You can also work with your Arista SE to determine the best design for your use case.
